acceptance/tests/windows/reboot_pending.rb

```diff
@@ -38,3 +38,27 @@
   #Verify that a shutdown has been initiated and clear the pending shutdown.
   retry_shutdown_abort(agent)
 end
+
+windows_agents.each do |agent|
+  original_name = nil
+
+  begin
+    step "Retrieve computer name"
+    original_name = on agent, 'cmd /c hostname'
+
+    new_name = ('a'..'z').to_a.shuffle[0, 12].join
+    step "Rename the computer to #{new_name} temporarily"
+    on agent, powershell("$(Get-WmiObject win32_computersystem).Rename('#{new_name}')")
+
+    step "Reboot if Pending Reboot Required"
+    apply_manifest_on agent,  reboot_manifest
+
+    #Verify that a shutdown has been initiated and clear the pending shutdown.
+    retry_shutdown_abort(agent)
+  ensure
+    if original_name
```

Comment:
Yeah, I think I did it like this because I was concerned the `on agent` could fail, but the rename could actually succeed.  Then we'd be in some trouble.

But maybe that doesn't happen in practice?